session2/3-challenge/7_array.rb

```diff
@@ -9,3 +9,9 @@
 # alternate_words("Can't we all get along?")      # => ["Can't", "all", "along"]
 # alternate_words("Elementary, my dear Watson!")  # => ["Elementary", "dear"]
 
+def alternate_words(sentence)
+	ret_arr = []
+	str_arr = sentence.gsub(/[!@$#%^&*()-=_+\[\]:;,.<>?\\|\/]/ ," ").split(" ")
```

Comment:
Would it not be easier to match just the words with regex? There is a mistake in one of the tests where "’" is used instead of "'" but once you account for that, it would save you the whole gsub operation.

Here's the regex I used to select all the relevant words:
(?!['|’].*['|’])\b[\w['|’]]+\b

Reply:
I suppose it may be fewer operations but a bit more challenging to understand. Nice regex though, you are a bit of a master with those. I know where I'll be coming for regex help during the course

Reply:
LOL - Google :)
Small amendment to Regex:
(?!'|’.*'|’)\b[\w'|’]+\b

Explanation is here:
https://regex101.com/r/WMDLy1/2/

Original source of info here:
https://stackoverflow.com/questions/27715581/how-do-i-handle-contractions-with-regex-word-boundaries-in-javascript

Reply:
Thanks, I've been trying to work it out and its been driving me crazy. Gonna have to practice my regex